src/services/UserService.ts

```diff
@@ -155,6 +158,38 @@ class UserService {
             points,
         };
     }
+
+    async getUserQuizzes(userId: ObjectId, chapterId: ObjectId): Promise<GetUserQuizzesResponse> {
+        const allQuizzes = await QuizModel.find({ chapterId });
+        const doneQuizzes = await QuizDoneModel.find({ userId });
+        const doneQuizzesIds = doneQuizzes.map(({ quizId }) => quizId.toString());
+
+        const quizzes = allQuizzes.map((quiz) => {
+            const { _id: quizId, title: quizTitle } = quiz;
+            const isDone = doneQuizzesIds.includes(quizId.toString());
+
+            return {
+                quizId,
+                quizTitle,
+                isDone,
+            };
+        });
+
+        return quizzes;
+    }
+
+    async getQuizQuestions(userId: ObjectId, quizId: ObjectId): Promise<GetQuizQuestionsResponse> {
+        const quizDone = await QuizDoneModel.findOne({ userId, quizId });
+        if (quizDone) throw new ApplicationError('Quiz already done', StatusCodes.CONFLICT);
```

Comment:
Zastanawiam się czy powinniśmy zabierać możliwość wykonania quizu raz jeszcze po jego ukończeniu? Trzeba by to przegadać z chłopakami, ew z panem duchem